implementation/src/main/java/io/smallrye/config/ConfigMappingProvider.java

```diff
@@ -1065,6 +1069,32 @@ private static Set<String> additionalMappedProperties(final Set<String> mappedPr
             mappedProperties.remove(propertyName);
         }
 
+        Set<String> envRoots = new HashSet<>();
```

Comment:
Roots are already contained in a set, do we really need another set here? Or we can just use a simple array, pre-sized? There are good reasons we could get any duplicate, after replacing non alphanumeric with underscores?

Saying that because both iteration and set constructions will be way cheaper with a simple array

Reply:
It maybe still worth to pre size it at least (in the worst case will be underutilized, but won't suffer been enlarged?)